File: Data Structures & Algorithms/sort-colors/submission-2.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int z = 0, o = 0;
        for (int i =0;i<nums.size();i++){
            if (nums[i]==1){
                if (z > o){
                    o =z;
                }
                swap(nums[i], nums[o++]);
            }else if (nums[i] == 0){
                swap(nums[i], nums[z++]);
                if(o!=0){
                    swap(nums[i],nums[o++]);
                }
            }
        }
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        mp = {}
        mp[0] = 1
        cnt = 0
        prefix = 0

        for num in nums:
            prefix += num
            if prefix -k in mp:
                cnt += mp[prefix - k]
            if prefix in mp:
                mp[prefix] += 1
            else:
                mp[prefix] = 1
        
        return cnt 